### ISSUE TITLE
Migrate to latest Angular (or React?)

### ISSUE THREAD

**Issue by JudahGabriel** [JudahGabriel/Chavah#17]
It's not sustainable long-term to use Angular 1.x. We need to migrate to the latest Angular, or alternately rewrite the UI in React. From a ease of port aspect, porting to Angular would be preferable.

**Comment by kdcllc** [comment]
I agree about porting the application to Angular2+ version. We have to clean up some of the AngularJs code first, for example moving away from storing data apart of window object and possibly localStorage.

There is also an option to run in mix mode and slowly upgrade the functionality.

**Comment by JudahGabriel** [comment]
Agreed. Perhaps that would be the first order of business: getting things in a good state to port to modern Angular.

I'm opened to using mixed mode to slowly upgrade if you feel it's best.

**Comment by kdcllc** [comment]
I am working on the separation of concern for an existing web api and angularjs in order to have api versioning and new SPA implementation based on Angular 6+. For the mobile apps I would like to utilize Ionic as described in this article:
https://auth0.com/blog/building-an-audio-player-app-with-ionic-angular-rxjs-and-ngrx/?utm_source=mybridge&utm_medium=blog&utm_campaign=read_more
